scripts/core/exp-tensor-map.yml

```diff
@@ -0,0 +1,213 @@
+    - type: uint32_t
+      name: PixelsPerColumn
+      desc: "[in] Number of pixels per column."
+    - type: const uint32_t*
+      name: ElementStrides
+      desc: "[in] Array containing traversal stride in each of the TensorRank dimensions."
+    - type: $x_exp_tensor_map_interleave_flags_t
+      name: Interleave
+      desc: "[in] Type of interleaved layout the tensor addresses"
+    - type: $x_exp_tensor_map_swizzle_flags_t
+      name: Swizzle
+      desc: "[in] Bank swizzling pattern inside shared memory"
+    - type: $x_exp_tensor_map_l2_promotion_flags_t
+      name: L2Promotion
+      desc: "[in] L2 promotion size."
+    - type: $x_exp_tensor_map_oob_fill_flags_t
+      name: OobFill
+      desc: "[in] Indicates whether zero or special NaN constant will be used to fill out-of-bounds elements."
+    - type: $x_exp_tensor_map_handle_t*
+      name: hTensorMap
+      desc: "[out] Handle of the tensor map object."
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "`TensorRank < 3`"
+
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Encode tensor map with tiled data"
+class: $xTensorMap
+name: EncodeTiledExp
+details:
+    - "Tiled map encode."
+params:
+    - type: $x_device_handle_t
+      name: hDevice
+      desc: "[in] Handle of the device object."
+    - type: $x_exp_tensor_map_data_type_flags_t
+      name: TensorMapType
+      desc: "[in] Data type of the tensor object."
+    - type: uint32_t
+      name: TensorRank
+      desc: "[in] Dimensionality of tensor; must be at least 3."
+    - type: void*
+      name: GlobalAddress
+      desc: "[in] Starting address of memory region described by tensor."
+    - type: const uint64_t*
+      name: GlobalDim
+      desc: "[in] Array containing tensor size (number of elements) along each of the TensorRank dimensions."
+    - type: const uint64_t*
+      name: GlobalStrides
+      desc: "[in] Array containing stride size (in bytes) along each of the TensorRank - 1 dimensions."
+    - type: const uint32_t*
+      name: BoxDim
+      desc: "[in] Array containing traversal box size (number of elments) along each of the TensorRank dimensions. Specifies how many elements to be traversed along each tensor dimension."
+    - type: const uint32_t*
+      name: ElementStrides
+      desc: "[in] Array containing traversal stride in each of the TensorRank dimensions."
+    - type: $x_exp_tensor_map_interleave_flags_t
+      name: Interleave
+      desc: "[in] Type of interleaved layout the tensor addresses"
+    - type: $x_exp_tensor_map_swizzle_flags_t
+      name: Swizzle
+      desc: "[in] Bank swizzling pattern inside shared memory"
+    - type: $x_exp_tensor_map_l2_promotion_flags_t
+      name: L2Promotion
+      desc: "[in] L2 promotion size."
+    - type: $x_exp_tensor_map_oob_fill_flags_t
+      name: OobFill
+      desc: "[in] Indicates whether zero or special NaN constant will be used to fill out-of-bounds elements."
```

Comment:
This is a lot of arguments and its not extensible. I think some or all of these should move into a properties struct.

Also, if using properties structs, would it be possible to expose this functionality in a single entry point with pNext chain for the differences?